modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxStateImpl.java

```diff
@@ -390,9 +369,7 @@ public class IgniteTxStateImpl extends IgniteTxLocalStateAdapter {
 
     /** {@inheritDoc} */
     @Override public void onTxEnd(GridCacheSharedContext cctx, IgniteInternalTx tx, boolean commit) {
-        for (int i = 0; i < activeCacheIds.size(); i++) {
-            int cacheId = activeCacheIds.get(i);
-
+        for (int cacheId : activeCacheIds()) {
```

Comment:
Can we avoid copying the array after each transaction?

Reply:
What do you mean by "after"?

Reply:
I mean the `onTxEnd` callback. It seems to be called on every tx.

Reply:
Still not got a problem/proposal